include/onnxruntime/core/optimizer/graph_transformer.h

```diff
@@ -33,35 +33,40 @@ class GraphTransformer {
   }
 
   /** Apply the in-place transformation defined by this transformer to the provided Graph instance.
-  @param[in] providers Optional - providers this transformer can be applied to
+  @param[in] compatible_provider_types Optional - providers this transformer can be applied to
   @param[out] modified Set to true if the Graph was modified.
   @returns Status with success or error information.
   */
-  common::Status Apply(Graph& graph, bool& modified, const std::vector<std::string>& providers = {}) const;
+  common::Status Apply(Graph& graph, bool& modified,
+                       const std::vector<std::string>& compatible_provider_types = {}) const;
 
  protected:
   /** Helper method to call ApplyImpl on any subgraphs in the Node. */
-  common::Status Recurse(Node& node, bool& modified, int graph_level) const {
+  common::Status Recurse(Node& node, bool& modified,
+                         const std::vector<std::string>& compatible_provider_types, int graph_level) const {
     int subgraph_level = ++graph_level;
     for (auto& entry : node.GetAttributeNameToMutableSubgraphMap()) {
       auto& subgraph = *entry.second;
-      ORT_RETURN_IF_ERROR(ApplyImpl(subgraph, modified, subgraph_level));
+      ORT_RETURN_IF_ERROR(ApplyImpl(subgraph, modified, compatible_provider_types, subgraph_level));
     }
 
     return Status::OK();
-  }
+  }  
 
  private:
   ORT_DISALLOW_COPY_ASSIGNMENT_AND_MOVE(GraphTransformer);
 
   // Apply the transform to the graph.
   // graph_level is 0 for the main graph, and is incremented when descending into the subgraph of a node.
+  // complatible_provider_types contains a list of provider types for which transformer can be applied.
```

Comment:
>complatible_provider_types [](start = 5, length = 26)

compatible_provider_types
May be good to say 'execution provider' in some places as not everyone will know that 'provider' means 'execution provider'